src/coreComponents/mesh/generators/VTKMeshGenerator.cpp

```diff
@@ -0,0 +1,640 @@
+{
+  vtkIntArray * attributeDataArray =  getAttributeDataArray();
+
+  auto countCell = [&](localIndex c, localIndex & numElem, std::map<int,localIndex> & regionsElem ) { // auto can be multiple types in the same function
+    numElem++;
+    if (attributeDataArray != nullptr)
+    {
+      int region = attributeDataArray->GetValue(c);
+      GEOSX_LOG_RANK_0_IF( region < 0,"Attribute value " << region << " not supported, please use value >=0 to describe regions");
+      regions.insert(region);
+      std::map< int, localIndex>::iterator it = regionsElem.find(region);
+      if( it == regionsElem.end())
+      {
+        regionsElem[region] = 1;
+      }
+      else
+      {
+        it->second++;
+      }
+    }
+  };
+
+
+  for( vtkIdType c = 0; c < m_vtkMesh->GetNumberOfCells(); c++)
+  {
+    if( m_vtkMesh->GetCellType(c) == VTK_HEXAHEDRON )
+    {
+      countCell(c, numHex, regionsHex);
+    }
+    else if( m_vtkMesh->GetCellType(c) == VTK_TETRA )
+    {
+      countCell(c, numTet, regionsTetra);
+    }
+    else if( m_vtkMesh->GetCellType(c) == VTK_WEDGE )
+    {
+      countCell(c, numWedge, regionsWedges);
+    }
+    else if( m_vtkMesh->GetCellType(c) == VTK_PYRAMID )
+    {
+      countCell(c, numPyr, regionsPyramids);
+    }
+    else if( m_vtkMesh->GetCellType(c) == VTK_TRIANGLE || m_vtkMesh->GetCellType(c) == VTK_QUAD )
+    {
+      if (attributeDataArray != nullptr)
+      {
+        surfaces.insert(attributeDataArray->GetValue(c));
+      }
+    }
+  }
+
+
+  // Communicate all the surfaces attributes to the ranks
+  const std::vector< int > surfaceVector(surfaces.cbegin(), surfaces.cend());
+  int numSurfaceId = surfaceVector.size();
+  int totalNbSurfaceId = 0;
+  std::vector< int > surfaceSizes( MpiWrapper::commSize());
+  MpiWrapper::allReduce( &numSurfaceId, &totalNbSurfaceId, 1, MPI_SUM,MPI_COMM_GEOSX );
+  MpiWrapper::allgather( &numSurfaceId, 1, surfaceSizes.data(), 1,MPI_COMM_GEOSX );
+  allSurfaces.resize(totalNbSurfaceId);
+  std::vector< int > displacements( MpiWrapper::commSize(),0);
+  for(unsigned int i = 1; i < displacements.size();i++)
+  {
+    displacements[i] = displacements[i-1] + surfaceSizes[i-1];
+  }
+  MpiWrapper::allgatherv<int, int>(surfaceVector.data(),  surfaceVector.size(), allSurfaces.data(),surfaceSizes.data(), displacements.data(), MPI_COMM_GEOSX);
+
+  std::sort(allSurfaces.begin(), allSurfaces.end());
+  auto last = std::unique( allSurfaces.begin(), allSurfaces.end());
+  allSurfaces.erase(last, allSurfaces.end());
+}
+
+std::vector< vtkDataArray * > VTKMeshGenerator::findArrayToBeImported()
+{
+  std::vector< vtkDataArray * > arrayToBeImported;
+  vtkCellData * cellData = m_vtkMesh->GetCellData();
+  for( int i = 0; i < cellData->GetNumberOfArrays(); i++)
+  {
+    vtkAbstractArray * curArray = cellData->GetAbstractArray(i);
+    // We support only float and double type. Good assumption?
```

Comment:
Should we instead specify explicitly the names of arrays to be imported in the XML file? I'm worried that VTK datasets may often contain too many fields useless for GEOSX that will end up being imported for no good reason.

Reply:
is the are database somewhere in GEOSX where all the reserved names are registered ?

Reply:
No, there's not, and ideally we would like to avoid maintaining such a database. Personally I think it should always be up to the user what gets imported and what doesn't (e.g. when the dataset has a temperature field and it's a recognized field name by one of GEOSX's physics solvers, but the user doesn't want it loaded for this problem). Whether we achieve this via explicit include-lists (with an option to include-all) or exclude-lists, I'd like for some flexibility to exist there, including remapping the field names which is something I think we have with PAMELA mesh generator.

With that said, the current implementation (importing all floating point arrays) is a good default for now and should not hold back this PR. At some point we will be revisiting the data import procedure (because it needs to be separated from the mesh generation step eventually), and it may be a good chance to define optimal use patterns.